crates/turborepo-paths/src/absolute_system_path.rs

```diff
@@ -311,6 +332,37 @@ impl AbsoluteSystemPath {
         rel.components().next() != Some(Utf8Component::ParentDir)
     }
 
+    /// relation_to_path does a lexical comparison of path components to
+    /// determine how this path relates to the given path. In the event that
+    /// the paths are the same, we return `Parent`, much the way that `contains`
+    /// would return `true`.
+    pub fn relation_to_path(&self, other: &Path) -> PathRelation {
```

Comment:
I didn't want to allocate to get to `&AbsoluteSystemPath` since this is a potentially pretty hot codepath, but it's probably worth double-checking if we can do that.

Reply:
Looks like there isn't any allocation except for on the error path which would mean a panic which I think is acceptable.